File: servers/Services/AcasService/Models/Material.cs
using System.ComponentModel.DataAnnotations;
using AcasService.Dev;

namespace AcasService.Models;

[DynamoDBEntity("MaterialTableName")]
public class Material
{
    [Key]
    public string Id { get; set; } = string.Empty;
    
    [Required]
    public string LecturerId { get; set; } = string.Empty;
    
    [Required]
    public string ClassroomId { get; set; } = string.Empty;
    
    [Required]
    public string Filename { get; set; } = string.Empty;
    
    public string FileUrl { get; set; } = string.Empty;
    
    public string Description { get; set; } = string.Empty;
    
    public bool IsDeleted { get; set; }
    
    public DateTime CreatedDate { get; set; }
}

File: mobile_app/lib/features/models/material.dart
class Material {
  final String id;
  final String lecturerId;
  final String classroomId;
  final String filename;
  final String fileUrl;
  final String description;
  final bool isDeleted;
  final String createdDate;

  Material({
    required this.id,
    required this.lecturerId,
    required this.classroomId,
    required this.filename,
    required this.fileUrl,
    required this.description,
    required this.isDeleted,
    required this.createdDate,
  });

  factory Material.fromJson(Map<String, dynamic> json) {
    return Material(
      id: json['id'] ?? '',
      lecturerId: json['lecturerId'] ?? '',
      classroomId: json['classroomId'] ?? '',
      filename: json['filename'] ?? '',
      fileUrl: json['fileUrl'] ?? '',
      description: json['description'] ?? '',
      isDeleted: json['isDeleted'] ?? false,
      createdDate: json['createdDate'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'lecturerId': lecturerId,
      'classroomId': classroomId,
      'filename': filename,
      'fileUrl': fileUrl,
      'description': description,
      'isDeleted': isDeleted,
      'createdDate': createdDate,
    };
  }
}

class ApiResponse<T> {
  final bool success;
  final String message;
  final T? dataResponse;

  ApiResponse({
    required this.success,
    required this.message,
    this.dataResponse,
  });

  factory ApiResponse.fromJson(
    Map<String, dynamic> json,
    T Function(dynamic)? fromJsonT,
  ) {
    return ApiResponse(
      success: json['success'] ?? false,
      message: json['message'] ?? '',
      dataResponse: fromJsonT != null && json['dataResponse'] != null
          ? fromJsonT(json['dataResponse'])
          : null,
    );
  }
}
